src/DataTableState.php

```diff
@@ -111,7 +111,7 @@ private function handleSearch(ParameterBag $parameters)
             $column = $this->dataTable->getColumn((int) $key);
             $value = $this->isInitial ? $search : $search['search']['value'];
 
-            if ($column->isSearchable() && '' !== trim($value) && null !== $column->getFilter() && $column->getFilter()->isValidValue($value)) {
+            if ($column->isSearchable() && ('' !== trim($value))) {
```

Comment:
This change is essential to implement other adapters. I created an adapter for DBAL but I need this change to do PR.